src/gtfs/read.rs

```diff
@@ -553,48 +670,112 @@ where
                     stop_point.stop_area_id = stop_area.id.clone();
                     stop_areas.push(stop_area);
-                    stop_point
+                    stop_point.clone()
                 } else {
-                    objects::StopPoint::from(stop)
+                    skip_fail!(stop.into())
                 };
                 stop_point.comment_links = comment_links;
                 stop_point.equipment_id = equipment_id;
                 stop_points.push(stop_point);
             }
             StopLocationType::StopArea => {
-                let mut stop_area = objects::StopArea::from(stop);
+                let mut stop_area: objects::StopArea = skip_fail!(stop.into());
                 stop_area.comment_links = comment_links;
                 stop_area.equipment_id = equipment_id;
                 stop_areas.push(stop_area);
             }
-            StopLocationType::StopEntrance => {
-                warn!("stop location_type = 2 not handled for the moment, skipping",)
-            }
-            StopLocationType::GenericNode => {
-                warn!("stop location_type = 3 not handled for the moment, skipping",)
-            }
-            StopLocationType::BoardingArea => {
-                warn!("stop location_type = 4 not handled for the moment, skipping",)
+            _ => {
+                let mut stop_location = skip_fail!(objects::StopLocation::from_gtfs_stop(stop));
+                stop_location.comment_links = comment_links;
+                stop_location.equipment_id = equipment_id;
+                stop_locations.push(stop_location);
             }
         }
     }
     let stoppoints = CollectionWithId::new(stop_points)?;
     let stopareas = CollectionWithId::new(stop_areas)?;
-    Ok((stopareas, stoppoints))
+    let stoplocations = CollectionWithId::new(stop_locations)?;
+    Ok((stopareas, stoppoints, stoplocations))
+}
+
+pub fn read_pathways<H>(file_handler: &mut H) -> Result<CollectionWithId<objects::Pathway>>
+where
+    for<'a> &'a mut H: FileHandler,
+{
+    let file = "pathways.txt";
+    let gtfs_stops = read_stops(file_handler)?;
+    let mut stops: HashMap<String, Stop> = HashMap::new();
+
+    for gtfs_stop in gtfs_stops {
+        stops.insert(gtfs_stop.id.clone(), gtfs_stop);
+    }
+
+    let (reader, _path) = file_handler.get_file_if_exists(file)?;
+    match reader {
+        None => {
+            info!("Skipping {}", file);
+            Ok(CollectionWithId::new(vec![])?)
+        }
+        Some(reader) => {
+            info!("Reading {}", file);
+            let mut rdr = csv::Reader::from_reader(reader);
+            let mut pathways = vec![];
+            for pathway in rdr.deserialize() {
+                let mut pathway: Pathway = skip_fail!(pathway.map_err(|e| format_err!("{}", e)));
+
+                let from_stop_point =
+                    skip_fail!(stops.get(&pathway.from_stop_id).ok_or_else(|| format_err!(
```

Comment:
Should we do this kind of check on the read? The `sanitize` should take care of that eventually so I feel like we could just read the file `pathways.txt`, not caring about the Stop ID references (isn't it what we do already with other files?). On top of that, it would allow you to use `read_utils::read_opt_collections()` instead of creating this custom `read_pathways` function.

Reply:
We already do this for the stop times and the sanitize don't log orphan reference.

Reply:
I made a PR to create some logging in the `sanitize` function (see https://github.com/CanalTP/transit_model/pull/399).